File: 2022.01.19/python/1978.py
n = int(input())
arr = list(map(int,input().split()))
max_num = max(arr)
result = 0
sosoo = [False for _ in range(max_num+1)]
sosoo[0] = sosoo[1] = True
for i in range(2,int(max_num**0.5)+1):
    j = 2
    while (i*j <=max_num):
        sosoo[i*j] = True
        j += 1
for num in arr:
    if not sosoo[num]:
        result += 1
print(result)

File: Boj.단계별/9_기본수학2/javascript/1978.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');
const n = Number(input[0]);
const arr = input[1].split(" ").map(Number);
const maxNum = Math.max(...arr);
const primeNumbers = new Array(maxNum+1);
let result = 0;
primeNumbers[0] = primeNumbers[1] = true;
for (let i = 2 ; i < parseInt(Math.sqrt(maxNum))+1 ; i++){
    j = 2;
    while (i*j <= maxNum){
        primeNumbers[i*j] = true;
        j++;
    }
}

for (let i = 0 ; i<arr.length ; i++){
    if (!primeNumbers[arr[i]]) result++;
}
console.log(result)